### PR TITLE
Refact for cmake

### PR DESCRIPTION
**Pull request by liangtie**
Ajust for using with cmake

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.14)
+
+include(cmake/prelude.cmake)
```
Comment (xakod) [inline CMakeLists.txt]:
```
list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
include(prelude.cmake)
```
 You can use include without relative paths with this small trick

Comment (liangtie) [reply CMakeLists.txt]: Awesome repository , helped me a lot

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,18 +1,18 @@
 cmake_minimum_required(VERSION 3.14)
-
-include(cmake/prelude.cmake)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
+include(${CMAKE_MODULE_PATH}/prelude.cmake)
```
Comment (xakod) [inline CMakeLists.txt]: try ```include(prelude)```, it should work

Comment (liangtie) [reply CMakeLists.txt]: It works, actually tricky

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  DESCRIPTION "simple MainWindow class implementation with frameless window and custom dark style. "
+  HOMEPAGE_URL "https://github.com/Jorgen-VikingGod/Qt-Frameless-Window-DarkStyle"
+  LANGUAGES CXX
+)
+
+include(project-is-top-level)
+include(variables)
+include(link_qt)
+
+qt5_add_resources(
```
Comment (xakod) [inline CMakeLists.txt]: What about qt6?

File: example/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.14)
+
+project(Qt-Frameless-Window-DarkStyleExamples CXX)
+
+include(../cmake/project-is-top-level.cmake)
```
Comment (xakod) [inline example/CMakeLists.txt]: Here can be includes without prefix :)

File: cmake/link_qt.cmake [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS ${QT_LIBRARIES_NAMES})
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS ${QT_LIBRARIES_NAMES})
+include_directories(%{Qt:QT_INSTALL_PREFIX})
+
+list(APPEND QT_LIBRARIES
+  Qt${QT_VERSION_MAJOR}::Core
+  Qt${QT_VERSION_MAJOR}::Gui
+  Qt${QT_VERSION_MAJOR}::Widgets
+)
```
Comment (xakod) [inline cmake/link_qt.cmake:22]: I had never seen such unordinarily way to link qt but why not

**Review by xakod — APPROVED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (liangtie) [reply CMakeLists.txt]: Got it

Comment (liangtie) [reply CMakeLists.txt]: Can you believe even now in 2022 our products have been targeting on Windows XP o(╥﹏╥)o

**Review by xakod — APPROVED** [review]

**Pull request closed without merging**